testserver/testserver_test.go

```diff
@@ -91,14 +88,11 @@ func newTenantDBForTest(t *testing.T, secure bool) (*sql.DB, func()) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	if err := ts.Start(); err != nil {
-		t.Fatal(err)
-	}
 	tenant, err := ts.(tenantInterface).NewTenantServer()
 	if err != nil {
 		t.Fatal(err)
 	}
-	if err := tenant.Start(); err != nil {
+	if err := tenant.WaitForInit(); err != nil {
```

Comment:
After changing the `ts.WaitForInit` call in `NewTenantServer` to `tenant.WaitForInit` you should be able to remove this.